includes/class-wc-payments-order-service.php

```diff
@@ -201,9 +203,10 @@ public function mark_payment_dispute_created( $order, $dispute_id, $reason ) {
 			return;
 		}
 
-		$this->update_order_status( $order, self::STATUS_ON_HOLD );
-		$this->add_dispute_created_note( $order, $dispute_id, $reason );
-		$order->save();
+		if ( $this->add_dispute_created_note( $order, $dispute_id, $reason ) ) {
+			$this->update_order_status( $order, self::STATUS_ON_HOLD );
+			$order->save();
+		}
```

Comment:
I'm curious though what you think about this approach though:

AFAICT most `add_*_note` methods (ex. `add_dispute_created_note`) only generate the note and add it to the order. Instead of doing that, maybe they could just return the note, with us having it within a variable and being able to check whether it exists already? Then we can decouple the check for uniqueness from the addition of the note, ex.:

```suggestion
		$note = $this->generate_dispute_created_note( $order, $dispute_id, $reason );

		if ( $this->order_note_exists( $order, $note ) ) {
			return;
		}

		$this->update_order_status( $order, self::STATUS_ON_HOLD );
		$order->save();
		$order->add_note( $note );
```

I know that this looks a bit longer and cumbersome, but I like the idea of separating concerns. This way other developers wouldn't be expected to know that `add_dispute_created_note` would return `false` if the note already exists.

Reply:
Yes, makes sense. I originally went to this route but I withdrew it as it's kinda long. I am going to make the change towards this idea as it has the benefits of separating concerns and reading code easier.

Reply:
>  I am going to make the change towards this idea as it has the benefits of separating concerns and reading code easier.

Done this!